run

```diff
@@ -1,6 +1,27 @@
 #!/bin/bash
 set -euo pipefail
 
+log() {
+    msg="$1"
+
+    logline="[$(pwd) $0] $msg\n"
+
+    printf "$logline" >&1
```

Comment:
this seems pretty aggressive. why log to both stdout and stderr?

Reply:
Agree normal logs don't need to go to stdout, but it helps to have errors go to both stdout and stderr so that you get the context of what normal events happened before / after the error. I can make a separate `log_error` for that.